core/src/View/Kokkos_ViewAlloc.hpp

```diff
@@ -139,132 +120,56 @@ struct ViewValueFunctor<DeviceType, ValueType, /*IsTrivial=*/false> {
     }
   }
 
-  void construct_shared_allocation() {
-    parallel_for_implementation<ConstructTag>();
-  }
-
-  void destroy_shared_allocation() {
-#ifdef KOKKOS_ENABLE_IMPL_VIEW_OF_VIEWS_DESTRUCTOR_PRECONDITION_VIOLATION_WORKAROUND
-    if constexpr (std::is_same_v<typename ExecSpace::memory_space,
-                                 Kokkos::HostSpace>)
-      for (size_t i = 0; i < n; ++i) (ptr + i)->~ValueType();
-    else
-#endif
-    {
-      parallel_for_implementation<DestroyTag>();
+  // Shortcut for zero initialization
+  void zero_memset_implementation() {
+    uint64_t kpID = 0;
+    if (Kokkos::Profiling::profileLibraryLoaded()) {
+      // We are not really using parallel_for here but using beginParallelFor
+      // instead of begin_parallel_for (and adding "via memset") is the best
+      // we can do to indicate that this is not supposed to be tunable (and
+      // doesn't really execute a parallel_for).
+      Kokkos::Profiling::beginParallelFor(
+          "Kokkos::View::initialization [" + name + "] via memset",
+          Kokkos::Profiling::Experimental::device_id(space), &kpID);
     }
-  }
 
-  // This function is to ensure that the functor with DestroyTag is instantiated
-  // This is a workaround to avoid "cudaErrorInvalidDeviceFunction" error later
-  // when the function is queried with cudaFuncGetAttributes
-  void functor_instantiate_workaround() {
-#if defined(KOKKOS_ENABLE_CUDA) || defined(KOKKOS_ENABLE_HIP) || \
-    defined(KOKKOS_ENABLE_SYCL) || defined(KOKKOS_ENABLE_OPENMPTARGET)
-    if (false) {
-      parallel_for_implementation<DestroyTag>();
+    (void)ZeroMemset(
+        space, Kokkos::View<ValueType*, typename DeviceType::memory_space,
+                            Kokkos::MemoryTraits<Kokkos::Unmanaged>>(ptr, n));
+
+    if (Kokkos::Profiling::profileLibraryLoaded()) {
+      Kokkos::Profiling::endParallelFor(kpID);
+    }
+    if (default_exec_space) {
+      space.fence("Kokkos::View::initialization via memset");
     }
-#endif
   }
```

Comment:
What happened to this function?

Reply:
I got rid of it to see if we still need it (last commit)

Reply:
I withdrew it.  I had tried on Summit with CUDA 12.2 and I would have sworn it did not segfault.